transform.ml

```diff
@@ -38,33 +42,45 @@ let as_opt_function (transformation : transform_instructions) : opt_function =
         annotations = None } in
       Some (Instr.replace_active_version func res)
 
-let as_opt_program (transformation : opt_function) : opt_prog =
+let as_opt_program (transform : opt_function) : opt_prog =
+  let transform' instrs =
+    match transform instrs with
+    | None -> false, instrs
+    | Some instrs -> true, instrs in
   fun prog ->
-    let changed, main = !! transformation prog.main in
+    let changed, main = transform' prog.main in
     let reduce (changed, functions) func =
-      let c, f = !! transformation func in
+      let c, f = transform' func in
       (changed||c, f::functions) in
     let changed, functions = List.fold_left reduce (changed, []) prog.functions in
     if changed then Some { functions; main; } else None
 
 let optimistic_as_opt_function (transformation : create_optimistic_version)
                                (cleanup : transform_instructions) : opt_function =
+  let cleanup ({instrs} as inp) =
+    match cleanup inp with
+    | None -> instrs
+    | Some instrs -> instrs in
   fun func ->
     match transformation func with
     | None -> None
     | Some version ->
-      let _, cleaned = !! cleanup version.instrs in
+      let inp = Analysis.as_analysis_input func version in
+      let cleaned =  cleanup inp in
       let version = { version with instrs = cleaned } in
       Some { func with body = version :: func.body }
 
 (* All available optimizations *)
-let cleanup_all_instrs =
-  (Transform_cleanup.remove_jmp &&&
-   (Transform_cleanup.remove_unused_label &&&
-    (Transform_cleanup.remove_unused_decl &&&
-     Transform_cleanup.remove_unreachable_code)))
-let minimize_liverange_instrs = Transform_liveness.minimize_liverange &&& Transform_cleanup.remove_unused_decl
-let const_prop_instrs = Transform_cleanup.remove_unused_decl &&& Transform_constantfold.const_prop
+let cleanup_all_instrs = combine_transform_instructions [
+    Transform_cleanup.remove_unreachable_code;
+    Transform_cleanup.remove_unused_decl;
+    Transform_cleanup.remove_jmp;]
```

Comment:
super-nitpicky (as in: no need to change): I prefer to have `]` on its own line because it makes nicer diff when we insert a new line at the end. (It's not just taste, optimizing for diff readability is an *objective* criterion.)